packages/zoe/tools/priceAuthorityRegistry.js

```diff
@@ -0,0 +1,138 @@
+// @ts-check
+
+import { E } from '@agoric/eventual-send';
+import { makeStore } from '@agoric/store';
+import { assert, details } from '@agoric/assert';
+
+import '../exported';
+
+/**
+ * @typedef {Object} Deleter
+ * @property {() => void} delete
+ */
+
+/**
+ * @typedef {Object} PriceAuthorityRegistryAdmin
+ * @property {(pa: ERef<PriceAuthority>, brandIn: Brand, brandOut: Brand)
+ * => Deleter} registerPriceAuthority Add a unique price authority for a given
```

Comment:
I'm still not sure about this design. What are the pros and cons compared to a priceAuthority only being for a specific pair? For instance, if multipoolAutoswap had multiple priceAuthorities rather than one. 

A con is having to explicitly pass in the brandIn, brandOut (or amountIn, etc) each time. The upside is that it's explicit in the call alone, what you mean. Hmm, I think this sounds good.

Reply:
Mainly, I want `agoric.priceAuthority` to be the single canonical location for all "official and supported" Agoric price authorities.  There may be thousands of pairs, not just the two.  If we didn't allow a price authority to dispatch based on its arguments, we would have to have a separate mechanism for registering all the exchange vectors.

I like being explicit because the client already needs to know the brands for the prices it requests.

Reply:
> Hmm, I think this sounds good.

Oh, are you referring to the design in this PR, or the one where we have a price authority only for a single vector (or pair of vectors)?

Reply:
I think the design in this PR is good. The above was my attempt to discuss it asynchronously :D